hub/core/chunk_engine/write.py

```diff
@@ -166,5 +151,28 @@ def _get_last_chunk(
     return "", memoryview(bytes())
 
 
-def _random_chunk_name() -> str:
+def _generate_chunk_name() -> str:
     return str(uuid1())
+
+
+def _min_chunk_ct_for_data_size(size: int) -> int:
+    """Calculates the minimum number of chunks in which data of given size can be fit."""
+    return ceil(size / CHUNK_MAX_SIZE)
+
+
+def _chunk_has_space(chunk: memoryview) -> bool:
+    """Returns whether the given chunk has space to take in more data."""
+    return len(chunk) > 0 and len(chunk) < CHUNK_MIN_TARGET
```

Comment:
Why `len(chunk)` > 0?

Reply:
chunk of size 0 means that the chunk doesn't exist, no point in merging it with incoming data (see line 93)